.github/api_guidelines.md

```diff
@@ -8,10 +8,130 @@ in slack channel [#forum-acs-api-design](https://redhat-internal.slack.com/archi
 ## Table of Contents
 
 - [Naming Guidelines](#naming-guidelines)
+  - [Decoupling](decoupling)
   - [gRPC Service Name](#grpc-service-name)
   - [gRPC Method Name](#grpc-method-name)
   - [gRPC Message Name](#grpc-message-name)
 
+## Decoupling
+
+While certain data may exist in the database for internal purposes or other functionalities, it is important to 
+exercise caution when deciding what data to make accessible through the API. The rule of thumb to follow is to **not 
+expose data structures defined for internal purposes at APIs**. Define a data structure for sole purpose of 
+API use, generally in the same package were the API is defined. This applies to all APIs including the 
+inter-component APIs.
+- All services defined in the package [/proto/api/v2](https://github.com/stackrox/stackrox/blob/master/proto/api/v2)
+**must not** import from the following packages:
+  - [/proto/storage](https://github.com/stackrox/stackrox/blob/master/proto/storage) 
+  - [/proto/internalapi](https://github.com/stackrox/stackrox/blob/master/proto/internalapi) 
+  - [/proto/api/v1](https://github.com/stackrox/stackrox/blob/master/proto/api/v1)
+- All new services and methods defined in the package [/proto/api/v1](https://github.com/stackrox/stackrox/blob/master/proto/api/v1) 
```

Comment:
Do we have a plan for existing services and methods defined in /proto/api/v1?  If we plan to keep them coupled to storage structs, can we state that explicitly?

Reply:
No plans yet but there is POC in the works. However, calling out that they can remain coupled discourages motivated engineers who want to decouple them.